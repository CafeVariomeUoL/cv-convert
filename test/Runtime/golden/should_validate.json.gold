[
{
    "age": 25,
    "name": "Sam",
    "heart_rate": [
        110,
        98,
        123
    ],
    "blood_pressure": [
        {
            "sys": 95,
            "dia": 65
        },
        {
            "sys": 105,
            "dia": 72
        },
        {
            "sys": 98,
            "dia": 68
        }
    ],
    "subject_id": 0,
    "i": 0
} , {
    "age": 30,
    "name": "Joe",
    "heart_rate": [
        130,
        95
    ],
    "blood_pressure": [
        {
            "sys": 115,
            "dia": 70
        },
        {
            "sys": 103,
            "dia": 66
        }
    ],
    "subject_id": 1,
    "i": 1
}
]