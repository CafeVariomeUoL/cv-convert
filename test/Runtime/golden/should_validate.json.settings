{
  "processFunction": "return row;",
  "tsInterface": "export default interface Row {\n    subject_id:     number;\n    name:           string;\n    age:            number;\n    heart_rate:     number[];\n    blood_pressure: BloodPressure[];\n    i:              number;\n}\n\nexport interface BloodPressure {\n    sys: number;\n    dia: number;\n}\n",
  "jsonSchema": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": {
      "Row": {
        "$ref": "#/definitions/default"
      },
      "BloodPressure": {
        "title": "BloodPressure",
        "type": "object",
        "properties": {
          "sys": {
            "type": "number",
            "title": "sys"
          },
          "dia": {
            "type": "number",
            "title": "dia"
          }
        },
        "additionalProperties": false,
        "required": [
          "dia",
          "sys"
        ]
      },
      "default": {
        "title": "default",
        "type": "object",
        "properties": {
          "subject_id": {
            "type": "number",
            "title": "subject_id"
          },
          "name": {
            "type": "string",
            "title": "name"
          },
          "age": {
            "type": "number",
            "title": "age"
          },
          "heart_rate": {
            "type": "array",
            "items": {
              "type": "number"
            },
            "title": "heart_rate"
          },
          "blood_pressure": {
            "type": "array",
            "items": {
              "$ref": "#/definitions/BloodPressure"
            },
            "title": "blood_pressure"
          },
          "i": {
            "type": "number",
            "title": "i"
          }
        },
        "additionalProperties": false,
        "required": [
          "age",
          "blood_pressure",
          "heart_rate",
          "i",
          "name",
          "subject_id"
        ]
      }
    },
    "$ref": "#/definitions/default"
  }
}