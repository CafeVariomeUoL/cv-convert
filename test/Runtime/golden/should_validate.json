[{
  "subject_id": 0,
  "name": "Sam",
  "age": 25,
  "heart_rate": [110, 98, 123],
  "blood_pressure": [{"sys":95, "dia": 65}, {"sys":105, "dia": 72}, {"sys":98, "dia": 68}]
},
{
  "subject_id": 1,
  "name": "Joe",
  "age": 30,
  "heart_rate": [130, 95],
  "blood_pressure": [{"sys":115, "dia": 70}, {"sys":103, "dia": 66}]
}]